sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriterContainer.scala

```diff
@@ -195,6 +200,11 @@ private[sql] abstract class BaseWriterContainer(
     serializableConf.value.set("mapred.task.id", taskAttemptId.toString)
     serializableConf.value.setBoolean("mapred.task.is.map", true)
     serializableConf.value.setInt("mapred.task.partition", 0)
+    for (c <- codec) {
```

Comment:
It might be more clear to use pattern match in order to avoid iteration like format.
```
codec match {
  case Some(c) => ///
  case None => ///
}
```